spec/api/api_put_spec.rb

```diff
@@ -0,0 +1,167 @@
+	end
+	
+	it 'verifies that last name can be edited' do
+		body[:lastName] = 'Stark'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that email can be edited' do
+		body['email'] = 'robb-stark@gameofthrones.com'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that password can be edited' do
+		body[:password] = 'rob-stark@gameofthrones.com'
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that phone number can be edited' do
+		body[:phone] = rand(100000).to_s
+		response = api_cl.update_user(body[:username], body)
+		expect(response.status).to eq(200)
+	end
+	
+	# Negative functional testing
+	# Username
+	it 'verifies that username can not be less than 2 symbols' do
+		body[:username] = 'u'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	# First name
+	it 'verifies that first name can not be less than 2 symbols' do
+		body[:firstName] = 'O'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that first name can not consist numbers' do
+		body[:firstName] = '1232151'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that first name can not consist symbols' do
+		body[:firstName] = '(*#&%&'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that first name can not start with low letter' do
+		body[:firstName] = 'robb'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that last name can not consist mixed data' do
+		body[:firstName] = 'robbA&%34%^'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	# Last name
+	it 'verifies that last name can not be less than 2 symbols' do
+		body[:lastName] = 'R'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that last name can not consist numbers' do
+		body[:lastName] = '123'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that last name can not consist symbols' do
+		body[:lastName] = '((($!'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that last name can not start with low letter' do
+		body[:lastName] = 'stark'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
+	
+	it 'verifies that first name can not consist mixed data' do
+		body[:lastName] = 'stark134@*$AA'
+		response = api_cl.update_user(body[:username], body)
+		!expect(response.status).to eq(200)
+	end
```

Comment:
duplication of checks. Consider using shared examples https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples